### PR TITLE
add --perf_args usage to jenkins performance runner scripts

### PR DESCRIPTION
**Pull request by apolcyn**
needed to use the perf flame graphs on more jenkins performance jobs

### CODE REVIEW

**Comment by apolcyn** [comment]
btw I just saw that these were the scripts called by `gRPC_performance_master`, `gRPC_performance_sweep`, and `gRPC_performance_experiment` jenkins jobs

**Comment by matt-kwong** [comment]
I think you covered all of the performance jobs we run, but I would let Nico or Jan confirm.

File: tools/jenkins/run_full_performance.sh [hunk 1]
```diff
@@ -67,6 +69,7 @@ tools/run_tests/run_performance_tests.py \
     --bq_result_table performance_test.performance_experiment_windows \
     --remote_worker_host grpc-performance-windows1 grpc-performance-windows2 \
     --xml_report report_windows.xml \
+    --perf_args "$PERF_ARGS" \
```
Comment (jtattermusch) [inline tools/jenkins/run_full_performance.sh]: these are running on windows, using "perf" command will probably break them.

**Comment by apolcyn** [comment]
For setting the `$PERF_ARGS` variable: yeah I was planning on setting these with a `PERF_ARGS` environment variable in the jenkins UI. I figured we could set up default parameters to `PERF_ARGS` on these jobs that might would generate flame graphs, but then we could also turn them off for specific runs by passing an empty string.
Note perf profiling doesn't get turned on if `--perf_args` is passed an empty string.

From what I've seen, running under perf has been pretty low overhead, maybe 5-10%

Maybe this should change to hard code the perf parameters? Or run perf on different jobs?

**Comment by apolcyn** [comment]
Sorry I'm actually thinking this will not work out very well, mostly because the `experiment` and `master` jobs are running jobs for other languages besides c++/go (what the flame graphs are most relevant for).

Thinking these flame graphs might want to stay on the `sweep` job, but be put on a new performance job specific to flame graphs that could cover master/experiment c++/go scenarios, if this sounds ok.

**Comment by jtattermusch** [comment]
Yeah, that sounds alright - I think it makes sense to make these benchmarks run where the sweep suite is running. We already have enough stuff running on the perf workers where multilang configurations are running.

**Comment by apolcyn** [comment]
Latest updates add flamegraph directory name option to the driver, for running multiple `run_performance_tests.py` on a jenkins job, and also rebase and squash.

PTAL, this should be ready for review

For the new jenkins job:
Added https://grpc-testing.appspot.com/job/gRPC_performance_flamegraphs/ jenkins job for c++ and go flame graphs, which could be put into the regular loop. Latest job in that (https://grpc-testing.appspot.com/job/gRPC_performance_flamegraphs/10/), has profiles separately for c++ and go.

**Comment by ctiller** [comment]
Is this still active?

**Comment by apolcyn** [comment]
Yes this should be ready. Actually the jenkins job created along with it is still alive in https://grpc-testing.appspot.com/job/gRPC_performance_flamegraphs/ - the most recent job there was has flame graphs for c++ and go attached. 

(btw the new job calls the new `tools/jenkins/run_performance_flamegraphs.sh` script added here)

**Comment by ctiller** [comment]
This is amazing. Please merge and enable it for sweeps.

**Comment by apolcyn** [comment]
failed tests were under `bazel full`, ignoring

log of failures in https://grpc-testing.appspot.com/job/gRPC_pull_requests_Bazel_full/38/console

**Comment by apolcyn** [comment]
@ctiller, done, these reports should start showing up in the sweep job

**Comment by apolcyn** [comment]
heads up on sweeps these aren't running yet - needed to re-run worker init scripts.

**Pull request merged**